include/openssl/bio.h

```diff
@@ -255,7 +255,8 @@ typedef struct bio_method_st BIO_METHOD;
 const char *BIO_method_name(const BIO *b);
 int BIO_method_type(const BIO *b);
 
-typedef void bio_info_cb(BIO *, int, const char *, int, long, long);
+typedef int BIO_info_cb(BIO *, int, int);
+typedef BIO_info_cb bio_info_cb;  /* backward compatibility */
```

Comment:
With the changed parameters, `BIO_info_cb` (and because it's an alias, `bio_info_cb` too) becomes an entirely different type.  Anyone who's made their own info callback is screwed.  *This is a no-can-do until 1.2.0*

Reply:
Unfortunately the this signature is how the callback is invoked.
Please see conn_state in bss_conn.c

Reply:
Anyone who's callback retunrs void, is in fact totally screwed:
```
        if (cb != NULL) {
            if ((ret = cb((BIO *)b, c->state, ret)) == 0)
                goto end;
        }
```

Reply:
Damnit...  yeahok, map versus reality, perhaps reality should win this one...